### PR TITLE
WIP feat(collector): improve remote host collector speed

### PR DESCRIPTION
**Pull request by DexterYan**
## Description, Motivation and Context

Please include a summary of the change or what problem it solves. Please also include relevant motivation and context.

## Checklist

- [ ] New and existing tests pass locally with introduced changes.
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] The commit message(s) are informative and highlight any breaking changes
- [ ] Any documentation required has been added/updated. For changes to https://troubleshoot.sh/ create a PR [here](https://github.com/replicatedhq/troubleshoot.sh/pulls)

## Does this PR introduce a breaking change?
- [ ] Yes
- [ ] No

### CODE REVIEW

File: pkg/supportbundle/collect.go [hunk 1]
```diff
@@ -207,136 +222,365 @@ func getAnalysisFile(analyzeResults []*analyze.AnalyzeResult) (io.Reader, error)
...
+				result := map[string]string{}
+				if err := json.Unmarshal(stdout, &result); err != nil {
+					return err
+				}
+
+				for file, data := range result {
+					results[file] = []byte(data)
+				}
+
+				time.Sleep(1 * time.Second)
```
Comment (marccampbell) [inline pkg/supportbundle/collect.go]: Why this sleep?

File: pkg/supportbundle/collect.go [hunk 2]
```diff
@@ -207,136 +222,365 @@ func getAnalysisFile(analyzeResults []*analyze.AnalyzeResult) (io.Reader, error)
...
+			}
+
+			// wait for log stream to catch up
+			time.Sleep(1 * time.Second)
```
Comment (marccampbell) [inline pkg/supportbundle/collect.go]: Sleep isn't great here. Better to wait for the object. This will be brittle

Comment (DexterYan) [reply pkg/supportbundle/collect.go]: Thank you! Agree, I am trying to use polling method. However, the previous [PR](https://github.com/replicatedhq/troubleshoot/pull/1671) will be refactored. This PR will be on hold.

File: pkg/supportbundle/collect.go [hunk 3]
```diff
@@ -207,136 +221,394 @@ func getAnalysisFile(analyzeResults []*analyze.AnalyzeResult) (io.Reader, error)
...
+	for _, pod := range pods.Items {
+		eg.Go(func() error {
+			// TODO: set timeout waiting
+			if err := waitForPodRunning(ctx, clientset, &pod); err != nil {
+				return err
+			}
+
+			results := map[string][]byte{}
+			for _, collectorSpec := range hostCollectors {
+				collector, ok := collect.GetHostCollector(collectorSpec, bundlePath)
```
Comment (diamonwiggins) [inline pkg/supportbundle/collect.go]: Is the reporting and the progress updates accurate if the flow is:

```
for pod in pods:
    run each collector
```

or should it be:

```
for collector in collectors:
    run in each pod
```

**Pull request closed without merging**